src/code/z_jpeg.c

```diff
@@ -1,11 +1,298 @@
+                    break;
+                }
+                case MARKER_SOS: {
+                    // Start of Scan marker, indicates the start of the image data.
+                    ptr += Jpeg_GetUnalignedU16(ptr);
+                    jpegCtx->imageData = ptr;
+                    break;
+                }
+                case MARKER_EOI: {
+                    // End of Image
+                    exit = true;
+                    break;
+                }
+                default: {
+                    ptr += Jpeg_GetUnalignedU16(ptr);
+                    break;
+                }
+            }
+        }
+    }
+}
+
+s32 Jpeg_Decode(void* data, void* zbuffer, void* work, u32 workSize) {
+    s32 y;
+    s32 x;
+    s32 j;
+    s32 i;
+    JpegContext jpegCtx;
+    JpegHuffmanTable hTables[4];
+    JpegDecoder decoder;
+    JpegDecoderState state;
+    JpegWork* workBuff = work;
+
+    if (workSize < sizeof(JpegWork)) {
+        return -1;
+    }
+
+    osCreateMesgQueue(&jpegCtx.mq, &jpegCtx.msg, 1);
+    MsgEvent_SendNullTask();
+
+    jpegCtx.workBuf = workBuff;
+
+    Jpeg_ParseMarkers(data, &jpegCtx);
+
+    switch (jpegCtx.dqtCount) {
+        case 1:
+            JpegUtils_ProcessQuantizationTable(jpegCtx.dqtPtr[0], &workBuff->qTableY, 3);
+            break;
+
+        case 2:
+            JpegUtils_ProcessQuantizationTable(jpegCtx.dqtPtr[0], &workBuff->qTableY, 1);
+            JpegUtils_ProcessQuantizationTable(jpegCtx.dqtPtr[1], &workBuff->qTableU, 1);
+            JpegUtils_ProcessQuantizationTable(jpegCtx.dqtPtr[1], &workBuff->qTableV, 1);
+            break;
+
+        case 3:
+            JpegUtils_ProcessQuantizationTable(jpegCtx.dqtPtr[0], &workBuff->qTableY, 1);
+            JpegUtils_ProcessQuantizationTable(jpegCtx.dqtPtr[1], &workBuff->qTableU, 1);
+            JpegUtils_ProcessQuantizationTable(jpegCtx.dqtPtr[2], &workBuff->qTableV, 1);
+            break;
+
+        default:
+            return -1;
+    }
+
+    switch (jpegCtx.dhtCount) {
+        case 1:
+            if (JpegUtils_ProcessHuffmanTable(jpegCtx.dhtPtr[0], &hTables[0], workBuff->codesLengths, workBuff->codes, 4)) {
+                return -1;
+            }
+            break;
+
+        case 4:
+            if (JpegUtils_ProcessHuffmanTable(jpegCtx.dhtPtr[0], &hTables[0], workBuff->codesLengths, workBuff->codes, 1)) {
+            } else if (JpegUtils_ProcessHuffmanTable(jpegCtx.dhtPtr[1], &hTables[1], workBuff->codesLengths,
+                                                     workBuff->codes, 1)) {
+            } else if (JpegUtils_ProcessHuffmanTable(jpegCtx.dhtPtr[2], &hTables[2], workBuff->codesLengths,
+                                                     workBuff->codes, 1)) {
+            } else if (!JpegUtils_ProcessHuffmanTable(jpegCtx.dhtPtr[3], &hTables[3], workBuff->codesLengths,
+                                                      workBuff->codes, 1)) {
+                break;
+            }
+            return -1;
```

Comment:
Er, what?

Reply:
This was the OoT version
```c

        case 4:
            if (JpegUtils_ProcessHuffmanTable(ctx.dhtPtr[0], &hTables[0], workBuff->codesLengths, workBuff->codes, 1)) {
                osSyncPrintf("Error : Cant' make huffman table.\n");
            }
            if (JpegUtils_ProcessHuffmanTable(ctx.dhtPtr[1], &hTables[1], workBuff->codesLengths, workBuff->codes, 1)) {
                osSyncPrintf("Error : Cant' make huffman table.\n");
            }
            if (JpegUtils_ProcessHuffmanTable(ctx.dhtPtr[2], &hTables[2], workBuff->codesLengths, workBuff->codes, 1)) {
                osSyncPrintf("Error : Cant' make huffman table.\n");
            }
            if (JpegUtils_ProcessHuffmanTable(ctx.dhtPtr[3], &hTables[3], workBuff->codesLengths, workBuff->codes, 1)) {
                osSyncPrintf("Error : Cant' make huffman table.\n");
            }
            break;
```

So, it had a bunch of printfs that aren't here anymore.
Should I add a comment here?

Reply:
Seems like a good idea, although I'm surprised it needed to be elses too.